keyboards/chromatonemini/keymaps/default/keymap.c

```diff
@@ -0,0 +1,110 @@
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+// Defines names for use in layer keycodes and the keymap
+enum layer_names {
+    _BASE,
+    _TRANS,
+    _FN
+};
+
+//  Don't change the DEFAULT_SCALE_COL value below. It must be 0.
+#define DEFAULT_SCALE_COL 0
+static uint8_t scale_indicator_col = DEFAULT_SCALE_COL;
+
+// Defines the keycodes used by our macros in process_record_user
+enum custom_keycodes {
+    SHIFTDN = SAFE_RANGE,
+    SHIFTUP,
+    TGLINDI,  //  ToGgLeINDIcator
+
+    L_BASE,
+    L_TRANS
+};
+
+// Long press: go to _FN layer, tap: MUTE
+#define FN_MUTE LT(_FN, KC_MUTE)
+
+// Used to set octave to MI_OCT_0
+extern midi_config_t midi_config;
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    /* Base */
+    [_BASE] = LAYOUT(
+            FN_MUTE,          MI_SUS,                            KC_VOLD, KC_VOLU,
+            MI_BENDU,
+        SHIFTDN, SHIFTUP,    MI_C_1, MI_D_1, MI_E_1,  MI_Fs_1, MI_Ab_1, MI_Bb_1, MI_C_2, MI_D_2, MI_E_2, MI_Fs_2, MI_Ab_2, MI_Bb_2, MI_C_3,  MI_D_3,  MI_E_3, MI_Fs_3, MI_Ab_3, MI_Bb_3, MI_C_4,
+            MI_BENDD,           MI_Db_1, MI_Eb_1, MI_F_1,  MI_G_1,  MI_A_1,  MI_B_1, MI_Db_2, MI_Eb_2, MI_F_2,  MI_G_2, MI_A_2,  MI_B_2, MI_Db_3, MI_Eb_3, MI_F_3,  MI_G_3,  MI_A_3,  MI_B_3
+    ),
+
+    /* Trans   This layer must locate 1 layer above _BASE layer. */
+    [_TRANS] = LAYOUT(
+            _______,          _______,                             _______, _______,
+            _______,
+        MI_TRNSD, MI_TRNSU,   _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______,
+            _______,               _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______
+    ),
+
+    [_FN] =  LAYOUT(
+            _______,          XXXXXXX,                             XXXXXXX, XXXXXXX,
+            MI_VELU,
+        MI_OCTD, MI_OCTU,     L_BASE, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX,
+            MI_VELD,               L_TRANS, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, XXXXXXX, RGB_SAD, RGB_HUD, RGB_SPD, RGB_VAD, RGB_RMOD, TGLINDI
+    )
+};
+
+void keyboard_post_init_user(void) {
+    //  Set otave to MI_OCT_0
+    midi_config.octave = MI_OCT_0 - MIDI_OCTAVE_MIN;
+
+    default_layer_set(1UL << _BASE);
```

Comment:
This shouldn't be needed, as it's run by default. 
```suggestion
```

Reply:
> This shouldn't be needed, as it's run by default.
@drashna 
The same discussion has been done in the previous pull request about this. Sorry for the confusion.
https://github.com/qmk/qmk_firmware/pull/12768#discussion_r652349388

By default, it will do `default_layer_set(0)` but what I want to do here is `default_layer_set(1)`.
It is indeed confusing but 
  `1UL << _BASE`
= `1UL << 0`
= 1
and it is not 0.
"0x01 left-shifted 0 times = 1".

As it is written [here](https://beta.docs.qmk.fm/using-qmk/software-features/feature_layers#functions-id-functions), you need to use `layer_mask` instead of `layer` for both `layer_state_set()` and `default_layer_set()` commands.

This is why when you want to set layer 0, you need to feed  
`1UL << _BASE` 
= `1UL << 0` 
= 1
to these functions.